### PR TITLE
Small profile-picture cleanups / fixes

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This fixes a very silly bug, where pictures were deleted twice. It looks like the code handled this gracefully, but still it looks scary.

### Changes made:

1) Improve debug messages: add \n.
2) Don't delete pictures twice.

### Release note:

Bug probably not severe enough.
### Documentation change:

No.
### Mentions:

@dirkhh

### CODE REVIEW

**Comment by bstoeger** [comment]
Added a commit that pushes `displayed_dive` from a helper class to the profile, with the ultimate goal of removing global state from the profile.

File: commands/command_pictures.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ void SetPictureOffset::redo()
 {
 	picture *pic = dive_get_picture(d, filename);
 	if (!pic) {
-		fprintf(stderr, "SetPictureOffset::redo(): picture disappeared!");
+		fprintf(stderr, "SetPictureOffset::redo(): picture disappeared!\n");
```
Comment (dirkhh) [inline commands/command_pictures.cpp:27]: my understanding was that `stderr` wasn't line buffered. This doesn't appear to be part of any standard, that seems to say it is implementation dependent, but I saw several references to this as "the convention" and lots of code appears to assume that this is indeed the case.
All that said, the new line makes this much more readable.

File: profile-widget/divepixmapitem.cpp [hunk 2]
```diff
@@ -5,13 +5,9 @@
 #include "core/qthelper.h"
 #include "core/settings/qPrefDisplay.h"
 #include "core/subsurface-qt/divelistnotifier.h"
-#ifndef SUBSURFACE_MOBILE
-#include "core/dive.h" // for displayed_dive
-#include "commands/command.h"
-#endif
```
Comment (dirkhh) [inline profile-widget/divepixmapitem.cpp:11]: typo in the commit message... "which know**S** which dive it represents"

**Review by dirkhh — APPROVED** [review]
This looks good, a couple small comments.
Feel free to merge after fixing the commit message.

**Comment by bstoeger** [comment]
Interesting thing about stderr vs. stdout. I guess that makes sense. Changed the commit messages - will merge.

**Pull request merged**